src/http/common.cr

```diff
@@ -82,7 +82,22 @@ module HTTP
     end
 
     io << "\r\n"
-    io << body if body
+
+    if body
+      if body.is_a?(IO)
+        if headers["Transfer-Encoding"] == "chunked"
```

Comment:
So if body is an IO and Transfer-Encoding is not chunked, we write no body at all? That doesn't seem right.